solstdlib/src/precompiles.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+
+function sha256(bytes memory input) returns (bytes32 ret) {
+  assembly {
+    let success := staticcall(gas(), 2, add(input, 32), mload(input), 0, 32)
+    if iszero(success) { revert(0, 0) }
```

Comment:
These precompiles currently differ two ways from the IR:
1) only support never EVM version where passing all gas is possible (would need #10283 to support conditional options)
2) it does not pass through the revert data (this should be fixed before merging)